### ISSUE TITLE
Expand CommandDefinition.OnCompleted to pass the DBCommand (allow for items like mysql last insert id to be fetched easier)

### ISSUE THREAD

**Issue by mitchcapper** [StackExchange/Dapper#748]
There is OnCompleted currently which executes after a command is finished.  It would be nice if we changed this handler to allow specifying a function to take the DbCommand itself.
This would be useful for things like MySQL which then have .LastInsertedId as part of the command available. 

This is in part a possible fix to #747 to allow it to use lasterinsertedid.  If this sounds like a good option I can generate a PR.

**Comment by NickCraver** [comment]
Is there another use case besides that issue? The plan is in V2 to change that .Insert pipeline to generic. It currently handles *only* integers which is a big fundamental problem. I'll make some breaking changes there to properly support all types (also see #722 for related items on columns) properly.

**Comment by mitchcapper** [comment]
Yeah I love #722 with columns I currently maintain a patched version of this per #385  ( https://github.com/mitchcapper/dapper-dot-net/compare/master...mitchcapper:additional_accessibility_helpers ) that also adds external helpers to help with mapping renamed columns,tables, and select queries when hand writing SQL.

I figured someone has a use case for oncompleted in the first place, it would seem to make sense to pass the DBCommand to the function incase the user wants to check something else for the result.

Technically I would love to hear another solution about how best to do this.  When I went hacking to fix the concurrency issue seeing the onCompleted call right there made it seem like a more proper way of doing so as one could do something like this then at a user level:
```
class SetId : IParameterCallbacks {
			public SetId( Action<long> on_completed_id) {
				this.on_completed_id = on_completed_id;
			}
			Action<long> on_completed_id;
			public void OnCompleted(DbCommand cmd) {
				on_completed_id((cmd as MySql.Data.MySqlClient.MySqlCommand).LastInsertedId);
			}
			public void AddParameters(IDbCommand command, Identity identity) { }
		}
```
then to use the class on an expected insert:
```
new SetId ((long id) => last_inserted_id = id);
```

It is still hacky, but works with keeping Dapper driver neutral but allow for each userlevel driver implementation to do what they want.
